### PR TITLE
Homepage UI buildout

### PR DESCRIPTION
**Pull request by ten1seven**
https://jdf-homepage-ui-buildout-blocking--pointlesscorp.netlify.app/

Completed UI buildout for the home page. There are a few visual differences mainly to improve accessibility and responsiveness:

* Links are either underlined by default or have an underline on hover/focus.
* Project card text now is black or white depending on the best contrast with the background color.
* The project grid is now fluid with grid rather than having stepped fixed widths. Cards are also uniform in height thanks to grid.
* The rotating part of the photo grid now respects prefersReducedMotion and stops rotating when hovering over it.
* X instead of Twitter.
* Otherwise it should be nearly identical.

CURRENT
Screen Shot 2024-10-23 at 09 55 14-fullpage

NEW
Screen Shot 2024-10-23 at 10 01 49-fullpage

### CODE REVIEW

**Review by andrewmosby — APPROVED** [review]
Since you mentioned a11y, the filter buttons aren't accessible via keyboard. I don't think this should hold you up, though. Unless you want it all in one merge.

**Comment by ten1seven** [comment]
@andrewmosby thanks for the reminder! I meant to circle back to this. Give it a try now.

**Comment by ten1seven** [comment]
@andrewmosby also I just updated the "radio buttons" to actually be radio buttons so the keyboard interaction works as expected for that control type.

**Comment by andrewmosby** [comment]
👍 👍

File: src/storyblok/components/ProjectList.astro [hunk 1]
```diff
@@ -1,48 +1,164 @@
...
+    case 'tools':
+      return iconStampTools;
+    case 'experiments':
+      return iconStampExperiments;
+    case 'stunts':
+      return iconStampStunts;
+    default:
+      return '';
+  }
+}
```
Comment (joshuapease) [inline src/storyblok/components/ProjectList.astro]: I tend to favor a config object over `switch` in this case. 

```suggestion
function getIconStamp(type) {
  const iconStamps = {
    tools: iconStampTools,
    experiments: iconStampExperiments,
    stunts: iconStampStunts
  };

  return iconStamps[type] ?? '';
}
```

File: src/storyblok/components/ProjectList.astro [hunk 2]
```diff
@@ -1,48 +1,164 @@
...
+<div
+  class="px-3"
+  x-data="{
+    showMore: $persist(false),
+    filters: $persist({
+      tools: true,
+      experiments: true,
+      stunts: true
+    }),
+    projects: JSON.parse(JSON.stringify(projects)),
```
Comment (joshuapease) [inline src/storyblok/components/ProjectList.astro]: I'm not sure if I'm seeing how `projects` is used within Alpine. 

I see astro looping over projects on line 111, but I don't think that's making use of what's here in Alpine.

**Review by joshuapease — APPROVED** [review]
👍

File: src/storyblok/components/ProjectList.astro (hunk 2 above)
Comment (ten1seven) [reply src/storyblok/components/ProjectList.astro]: 👍 I must have forgotten to remove this when refactoring. Thanks for the catch!

**Comment by ten1seven** [comment]
Merged

**Pull request closed without merging**